src/web/components/Core/Loading.tsx

```diff
@@ -1,3 +1,12 @@
-export function Loading() {
-  return <div>Loading...</div>;
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type LoadingProps = Readonly<{
+  innerText?: string;
+}>;
+export function Loading({ innerText = 'Loading...' }: LoadingProps) {
+  return (
+    <div>
+      <FontAwesomeIcon icon='spinner' spin /> {innerText}
```

Comment:
I didn't end up using the Radix spinner - it's a premium-only component (and we're on the free open source version) and it includes a lot of functionality that's not relevant.